src/core/src/op/topk.cpp

```diff
@@ -403,24 +335,99 @@ bool op::v1::TopK::has_evaluate() const {
     return true;
 }
 
+bool op::v1::TopK::evaluate_lower(const HostTensorVector& output_values) const {
+    OV_OP_SCOPE(v1_TopK_evaluate_lower);
+    return get_input_tensor(1).has_and_set_bound() && default_lower_bound_evaluator(this, output_values);
+}
+
+bool op::v1::TopK::evaluate_upper(const HostTensorVector& output_values) const {
+    OV_OP_SCOPE(v1_TopK_evaluate_upper);
+    return get_input_tensor(1).has_and_set_bound() && default_upper_bound_evaluator(this, output_values);
+}
+
+bool op::v1::TopK::evaluate_label(TensorLabelVector& output_labels) const {
+    OPENVINO_ASSERT(output_labels.size() == get_output_size());
+
+    const auto& input_labels = get_input_tensor(0).get_value_label();
+
+    if (!has_no_labels(input_labels) && get_input_tensor(1).has_and_set_bound()) {
+        auto get_labels_by_idx = [this, &input_labels](const ov::Tensor& indicies) {
+            auto idxs = std::vector<uint64_t>();
+
+            switch (m_index_element_type) {
+            case element::i32:
+                idxs.insert(idxs.end(), indicies.data<int32_t>(), indicies.data<int32_t>() + indicies.get_size());
+                break;
+            case element::i64:
+                idxs.insert(idxs.end(), indicies.data<int64_t>(), indicies.data<int64_t>() + indicies.get_size());
+                break;
+            default:
+                OPENVINO_ASSERT(false, "Index element type has not supported type: ", m_index_element_type);
+            };
+
+            auto out = TensorLabel();
+            out.reserve(idxs.size());
+            for (const auto& i : idxs) {
+                out.emplace_back(input_labels[i]);
+            }
+            return out;
+        };
+
+        // get labels by top k indexes for lower and upper bound
+        ov::TensorVector lowers(get_output_size()), uppers(get_output_size());
+        auto lb_labels = evaluate_lower(lowers) ? get_labels_by_idx(lowers.back()) : ov::TensorLabel();
+        auto ub_labels = evaluate_upper(uppers) ? get_labels_by_idx(uppers.back()) : ov::TensorLabel();
+        // align sizes of labels
+        const auto common_size = std::max(lb_labels.size(), ub_labels.size());
+        lb_labels.resize(common_size, ov::no_label);
+        ub_labels.resize(common_size, ov::no_label);
+        // set labels for values output favorite the upper bound label
```

Comment:
Let's say there is `ParialShape{{2, 12}, {3, 10}, {4, 11}}` with overlapping interval dimension and labels {a, b, c}.
The final compatible static shape can be `Shape{4, 5, 6}` as well as `Shape{6, 5, 4}` or `Shape{5, 6, 4}` 
(...and other combinations).

The values taken as lower are: `{2, 3, 4}`, and upper `{12, 11, 10}` 
so the corresponding indices values for max top3 are:
indices lb: {2, 1, 0} => labels {c, b, a}
indices ub: {0, 2, 1} => labels {a, c, b}

for Shape{5, 6, 4} max top3 indices are {1, 0, 2} and the labels should be {b, a, c}.

So the proposed approach seems to be not universal for labels evaluation.

Reply:
Yes that is true, as I've used approach takes uppers and lowers and sort separately but it can result with illegal dimension or when static values will be used order will not be correct.

So maybe use other approach like:
As upper bound is usually more important so do evaluate for uppers only and then take their lowers and labels, and sort it that way. 
But if there is any overlapping dimension in this set then make output dynamic shape as order cannot be defined until static values will be used.

Example:
PartialShape{{2,3}, {10,12}, {1,4}, {5, 6}}
k = 4, max -> dynamic as there is overlap between {2,3}, and {1,4}
k = 3, max -> dynamic or there is overlap and indices cannot be specified for 3rd dimension
k = 2, max -> {{10, 12}, {5,6}} no overlaps
k = 4, min -> dynamic
k  = 3, min -> dynamic
k = 1, min -> dynamic 

Then:
Evaluate upper will do evaluation for uppers bounds but check corresponding lower for overlap if there is no over lap return true, and output tensor will be filled with uppers.

Evaluate lower will also do evaluation for uppers but fill with lower for corresponding topk uppers and return true if there is no overlap


@jane-intel @mitruska Share opinion about this solution or should I do it in another way.

Reply:
Implementation changed.

Reply:
The solution to return `false` for overlapping dimensions sounds better to me.
But in general the TopK in ShapeOf subgraph seems to be kind of theoretical case.

@jane-intel Could you confirm that the implementation of evaluate `evaluate_lower/upper/label` is applicable for TopK operation?

Reply:
Right now removed it from this PR and if bound evaluation required will provided as other PR.